pkg/transformers/namespace.go

```diff
@@ -97,5 +97,49 @@ func (o *namespaceTransformer) Transform(m resmap.ResMap) error {
 		}
 
 	}
+	o.updateClusterRoleBinding(m)
 	return nil
 }
+
+func (o *namespaceTransformer) updateClusterRoleBinding(m resmap.ResMap) {
+	samap := resmap.ResMap{}
+	sagvk := schema.GroupVersionKind{Group: "core", Version: "v1", Kind: "ServiceAccount"}
+	for id := range m {
+		if id.Gvk().String() == sagvk.String() {
+			samap[id] = m[id]
+		}
+	}
+
+	for id := range m {
+		if id.Gvk().Kind != "ClusterRoleBinding" && id.Gvk().Kind != "RoleBinding" {
+			continue
+		}
+		objMap := m[id].UnstructuredContent()
+		subjects := objMap["subjects"].([]interface{})
+		for i := range subjects {
+			subject := subjects[i].(map[string]interface{})
+			kind, foundk := subject["kind"]
+			name, foundn := subject["name"]
+			if !foundk || !foundn || kind.(string) != "ServiceAccount" {
+				continue
+			}
+			if name == "default" {
```

Comment:
we should add comment explaining why "default" account is being treated as a kustomized managed service account.